lib/Sema/CSSimplify.cpp

```diff
@@ -15201,7 +15201,16 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
                locator.endsWith<LocatorPathElt::SubscriptMember>())
                   ? 1
                   : 0;
-
+    // An overload choice that isn't settable is least interesting for diagnosis.
+    if (auto overload = findSelectedOverloadFor(getCalleeLocator(fix->getLocator()))) {
+      if (auto decl = overload->choice.getDeclOrNull()) {
```

Comment:
All of the nested conditionals could be simplified down to:
```c++
if (auto *var = dyn_cast_or_null<VarDecl>(overload->choice.getDeclOrNull())) {
   impact += !var->isSettableInSwift(DC) ? 1 : 0;
}
```

Reply:
Thanks, done!